pkg/serverless/logs/logs.go

```diff
@@ -275,8 +280,14 @@ func processLogMessages(c *LambdaLogsCollector, messages []logMessage) {
 				continue
 			}
 			ecs := c.ExecutionContext.GetCurrentState()
-			logMessage := logConfig.NewChannelMessageFromLambda([]byte(message.stringRecord), message.time, ecs.ARN, ecs.LastRequestID)
-			c.LogChannel <- logMessage
+			if len(message.objectRecord.requestID) > 0 {
+				if message.objectRecord.requestID != ecs.LastRequestID {
+					log.Warnf("Mismatched requestID between %s and %s", message.objectRecord.requestID, ecs.LastRequestID)
+				}
+				c.LogChannel <- logConfig.NewChannelMessageFromLambda([]byte(message.stringRecord), message.time, ecs.ARN, message.objectRecord.requestID)
+			} else {
+				c.LogChannel <- logConfig.NewChannelMessageFromLambda([]byte(message.stringRecord), message.time, ecs.ARN, ecs.LastRequestID)
+			}
```

Comment:
This is a nit pick. I want to share another option of how this could be structured that could be both easier to support long term and require a smaller diff. You're welcome to ignore this (and any of my other nit picks) since the code works fine and great as currently written.

The only thing that is different between the if and else blocks is where we get the request ID. Therefore, instead of copying all of the code into two locations, just set the request ID in this block.

```go
var requestID string
if message.objectRecord.requestID != "" {
    if message.objectRecord.requestID != ecs.LastRequestID {
        log.Warnf("Mismatched requestID between %s and %s", message.objectRecord.requestID, ecs.LastRequestID)
    }
    requestID = message.objectRecord.requestID
} else {
    requestID = ecs.LastRequestID
}

logMessage := logConfig.NewChannelMessageFromLambda([]byte(message.stringRecord), message.time, ecs.ARN, requestID)
c.LogChannel <- logMessage
```